src/integrationTest/java/org/opensearch/security/DlsIntegrationTests.java

```diff
@@ -517,4 +552,92 @@ public void testAggregateAndComputeStarRatings() throws IOException {
             assertThat(((ParsedAvg) actualAggregation).getValue(), is(1.5));
         }
     }
+
+    @Test
+    public void testGetDocumentWithDLSRestrictions() throws Exception {
+        GetRequest findExistingDoc = new GetRequest(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1);
+        GetRequest findNonExistingDoc = new GetRequest(FIRST_INDEX_NAME, "RANDOM_INDEX");
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(BOOL_USER)) {
+            GetResponse boolFoundGetResponse = restHighLevelClient.get(findExistingDoc, DEFAULT);
+            assertThat(boolFoundGetResponse, containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1));
+
+            GetResponse boolNotFoundGetResponse = restHighLevelClient.get(findNonExistingDoc, DEFAULT);
+            assertThat(boolNotFoundGetResponse.isExists(), equalTo(false));
+        }
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(TERM_USER)) {
+            GetResponse termFoundGetResponse = restHighLevelClient.get(findExistingDoc, DEFAULT);
+            assertThat(termFoundGetResponse, containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1));
+
+            GetResponse termNotFoundGetResponse = restHighLevelClient.get(findNonExistingDoc, DEFAULT);
+            assertThat(termNotFoundGetResponse.isExists(), equalTo(false));
+        }
+    }
```

Comment:
```suggestion
public void testGetDocumentWithDLSRestrictions() throws Exception {
        GetRequest findExistingDoc = new GetRequest(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1);
        GetRequest findNonExistingDoc = new GetRequest(FIRST_INDEX_NAME, "RANDOM_INDEX");

        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(BOOL_USER)) {
            GetResponse boolFoundGetResponse = restHighLevelClient.get(findExistingDoc, DEFAULT);
            assertThat(boolFoundGetResponse, containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1));

            GetResponse boolNotFoundGetResponse = restHighLevelClient.get(findNonExistingDoc, DEFAULT);
            assertThat(boolNotFoundGetResponse.isExists(), equalTo(false));
        }

        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(TERM_USER)) {
            
        }
    }
    
    private void assertNotFoundGetResponse() {
            GetResponse response = restHighLevelClient.get(findExistingDoc, DEFAULT);
            assertThat(response, containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1));

            response = restHighLevelClient.get(findNonExistingDoc, DEFAULT);
            assertThat(response.isExists(), equalTo(false));
    }
```

This suggestion can be applied to all tests below for better readability and cleaner code.